core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -2271,36 +2299,40 @@ object Parser {
       }
     }
 
-    case class Rep0[A, B](p1: Parser[A], acc: Accumulator0[A, B]) extends Parser0[B] {
+    case class Rep0[A, B](p1: Parser[A], maxMinusOne: Int, acc: Accumulator0[A, B])
+        extends Parser0[B] {
       private[this] val ignore: B = null.asInstanceOf[B]
 
       override def parseMut(state: State): B = {
         if (state.capture) {
           val app = acc.newAppender()
-          if (repCapture(p1, 0, state, app)) app.finish()
+          if (repCapture(p1, 0, maxMinusOne, state, app)) app.finish()
           else ignore
         } else {
-          repNoCapture(p1, 0, state)
+          repNoCapture(p1, 0, maxMinusOne, state)
           ignore
         }
       }
     }
 
-    case class Rep[A, B](p1: Parser[A], min: Int, acc1: Accumulator[A, B]) extends Parser[B] {
+    /** A parser that can repeats the underlying parser multiple times
+      */
+    case class Rep[A, B](p1: Parser[A], min: Int, maxMinusOne: Int, acc1: Accumulator[A, B])
+        extends Parser[B] {
       if (min < 1) throw new IllegalArgumentException(s"expected min >= 1, found: $min")
 
       private[this] val ignore: B = null.asInstanceOf[B]
 
       override def parseMut(state: State): B = {
         val head = p1.parseMut(state)
-
+        def maxRemaining = if (maxMinusOne == Int.MaxValue) Int.MaxValue else maxMinusOne - 1
```

Comment:
Can we add a comment that this is to deal with the fact that we have already parsed one item, so if we are parsing a finite number we need to decrease by one.